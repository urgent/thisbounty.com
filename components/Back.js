import React from "react";
import { Grid, Image, Text, Box } from "@chakra-ui/react";
import Link from "next/link";

export default function Back({ href, text, id }) {
  return (
    <Box
      borderY="55px solid transparent"
      borderRight="20px solid"
      borderRightColor={"brand.400"}
      cursor="pointer"
      height="130px"
    >
      <Link href={href}>
        <Text
          data-cy={`back-${id}`}
          fontSize="xs"
          position="absolute"
          transform="rotate(-90deg)"
          fontWeight="bold"
        >
          {text}
        </Text>
      </Link>
    </Box>
  );
}
