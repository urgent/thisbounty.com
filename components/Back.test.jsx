import React from "react";
import renderer from "react-test-renderer";
import Back from "./Back";

it("renders correctly", () => {
  const tree = renderer
    .create(<Back href={"/bounty/work"} title="Back" id="work" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
