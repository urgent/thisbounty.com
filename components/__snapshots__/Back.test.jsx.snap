// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1n533yk"
>
  <p
    className="chakra-text css-1ykves1"
    data-cy="back-work"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
</div>
`;
